syntax = "proto3";

package mydb;

message Request {
    enum REQUEST_TYPE {
        INSERT_TEXT = 0;
        INSERT_LONG = 1;
        READ_TEXT = 2;
        READ_LONG = 3;
        UPDATE_TEXT = 4;
        UPDATE_LONG = 5;
        DELETE = 6;
        START_TRANSACTION = 7;
        COMMIT = 8;
        ABORT = 9;
    }
    REQUEST_TYPE type = 1;
    string table = 2;
    string key = 3;
    repeated string fields = 4;
    map<string, bytes> text_row = 5;
    string long_field = 6;
    sint64 long_row = 7;
}

message Response {
    enum RESPONSE_TYPE {
        READ_TEXT = 0;
        READ_LONG = 1;
        STATUS = 2;
    }
    RESPONSE_TYPE type = 1;
    bool isStatusOk = 2;
    string long_result = 3;
    map<string, bytes> text_result = 4;
}